src/picknik_ur_base_config/description/environment.xacro

```diff
@@ -0,0 +1,110 @@
+        </geometry>
+      </visual>
+      <collision>
+        <origin rpy="0 0 0" xyz="0 0 0"/>
+        <geometry>
+          <mesh filename="package://picknik_accessories/descriptions/furniture/generic_table/wood_block.dae" scale="0.25 0.25 0.1"/>
+        </geometry>
+      </collision>
+    </link>
+
+    <joint name="table_joint" type="fixed">
+      <parent link="base" />
+      <child link="environment" />
+      <origin rpy="0 0 0" xyz="0 0 -0.05" />
+    </joint>
+
+    <link name="box_1">
+      <xacro:cube_visual length="0.05" width="0.05" height="0.05">
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <color rgba="0.1 0.1 0.1 0.9" />
+      </xacro:cube_visual>
+    </link>
+
+    <joint name="box1_on_table" type="fixed">
+      <parent link="environment" />
+      <child link="box_1" />
+      <origin rpy="0 0 0" xyz="0.65 -0.35 0.1" />
+    </joint>
+
+    <link name="box_2">
+      <xacro:cube_visual length="0.05" width="0.05" height="0.05">
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <color rgba="0.5 0.5 0.5 0.9" />
+      </xacro:cube_visual>
+    </link>
+
+    <joint name="box2_on_table" type="fixed">
+      <parent link="environment" />
+      <child link="box_2" />
+      <origin rpy="0 0 0" xyz="0.65 0.35 0.1" />
+    </joint>
+
+    <link name="box_3">
+      <xacro:cube_visual length="0.05" width="0.05" height="0.05">
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <color rgba="0.8 0.3 0.1 0.9" />
+      </xacro:cube_visual>
+    </link>
+
+    <joint name="box3_on_table" type="fixed">
+      <parent link="environment" />
+      <child link="box_3" />
+      <origin rpy="0 0 0" xyz="0.65 0 0.1" />
+    </joint>
+
+    <link name="box_4">
+      <xacro:cube_visual length="0.05" width="0.05" height="0.05">
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <color rgba="0.8 0.3 0.1 0.9" />
+      </xacro:cube_visual>
+    </link>
+
+    <joint name="box4_on_table" type="fixed">
+      <parent link="environment" />
+      <child link="box_4" />
+      <origin rpy="0 0 0" xyz="-0.65 0.25 0.1" />
+    </joint>
+
+    <link name="box_5">
+      <xacro:cube_visual length="0.05" width="0.05" height="0.05">
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <color rgba="0.1 0.3 0.8 0.9" />
+      </xacro:cube_visual>
+    </link>
+
+    <joint name="box5_on_table" type="fixed">
+      <parent link="environment" />
+      <child link="box_5" />
+      <origin rpy="0 0 0" xyz="-0.65 -0.25 0.1" />
+    </joint>
+
+    <!-- External Camera -->
+    <link name="external_camera_link" />
+    <joint name="external_camera_joint" type="fixed">
+      <parent link="${parent}" />
+      <child link="external_camera_link" />
+      <origin xyz="-0.3 0.3 1.0" rpy="0.0 0.4 0" />
+    </joint>
+
+    <xacro:realsense_d435 parent="external_camera_link" name="scene_camera" visible="false">
+      <origin xyz="0 0 0" rpy="0 0 0" />
+    </xacro:realsense_d435>
+  </xacro:macro>
```

Comment:
I feel mixed on this one. For example, we might have an environment for our motion capture lab that would have many cameras. In that case, I would say the cameras are part of the environment. It would be easier to put a new robot in the same environment, without repeating the whole camera setup. For one camera, it seems easier to comprehend at the top level.

Reply:
Yeah, I conceptually agree, but was thinking more about separating the passive environment (collision objects, etc.) with the placement of external sensors.

It's also more visible to the user that you are explicitly putting a camera somewhere instead of it "just being part of the background".

That's why I was leaning top-level. But I defer to you!

Reply:
Okay, I think it can be changed easy enough in the future if needed, so I will leave it for now